diff.go

```diff
@@ -21,7 +21,7 @@ type KVPairReceiver func(pair *KVPair) error
 //
 // The algorithm don't run in constant memory strictly, but it tried the best the only
 // keep minimal intermediate states in memory.
-func (ndb *nodeDB) extractStateChanges(prevVersion int64, prevRoot *NodeKey, root *NodeKey, receiver KVPairReceiver) error {
+func (ndb *nodeDB) extractStateChanges(prevVersion int64, prevRoot, root []byte, receiver KVPairReceiver) error {
```

Comment:
I'm a big fan of this change in particular. Actually it's what I originally recommended. https://github.com/cosmos/iavl/pull/676#pullrequestreview-1329237749